pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/SchemaChangeHandler.java

```diff
@@ -0,0 +1,341 @@
+          } else {
+            throw new UnsupportedOperationException(
+                "Adding time column is not supported in schema evolution");
+          }
+        }
+      } else {
+        if (fieldSpec.getFieldType() == FieldType.DIMENSION) {
+          schemaUpdateActionMap.put(column, Action.ADD_DIMENSION);
+        } else if (fieldSpec.getFieldType() == FieldType.METRIC) {
+          schemaUpdateActionMap.put(column, Action.ADD_METRIC);
+        } else {
+          throw new UnsupportedOperationException(
+              "Adding time column is not supported in schema evolution");
+        }
+      }
+    }
+    return schemaUpdateActionMap;
+  }
+
+  @SuppressWarnings("unchecked")
+  private void createColumnWithDefaultValue(String column,
+      PropertiesConfiguration metadataProperties, File tmpIndexDir) throws Exception {
+    int totalDocs = metadata.getTotalDocs();
+    int totalRawDocs = metadata.getTotalRawDocs();
+    int totalAggDocs = metadata.getTotalDocs() - metadata.getTotalRawDocs();
+    // we need only 1 bit since we have only 1 value in the forward index
+    FieldSpec fieldSpec = schema.getFieldSpecFor(column);
+    int dictionaryElementSize;
+    Object min = fieldSpec.getDefaultNullValue();
+    Object max = fieldSpec.getDefaultNullValue();
+    Object sortedArray;
+    switch (fieldSpec.getDataType()) {
+    case STRING:
+    case BOOLEAN:
+      byte[] bytes = fieldSpec.getDefaultNullValue().toString().getBytes();
+      dictionaryElementSize = bytes.length;
+      sortedArray = new String[] {
+          fieldSpec.getDefaultNullValue().toString()
+      };
+      break;
+    case SHORT:
+      sortedArray = new short[] {
+          Short.valueOf(fieldSpec.getDefaultNullValue().toString())
+      };
+      dictionaryElementSize = fieldSpec.getDataType().size();
+      break;
+    case INT:
+      sortedArray = new int[] {
+          Integer.valueOf(fieldSpec.getDefaultNullValue().toString())
+      };
+      dictionaryElementSize = fieldSpec.getDataType().size();
+      break;
+    case FLOAT:
+      sortedArray = new float[] {
+          Float.valueOf(fieldSpec.getDefaultNullValue().toString())
+      };
+      dictionaryElementSize = fieldSpec.getDataType().size();
+      break;
+    case DOUBLE:
+      sortedArray = new Double[] {
+          Double.valueOf(fieldSpec.getDefaultNullValue().toString())
+      };
+      dictionaryElementSize = fieldSpec.getDataType().size();
+      break;
+    case LONG:
+      sortedArray = new Long[] {
+          Long.valueOf(fieldSpec.getDefaultNullValue().toString())
+      };
+      dictionaryElementSize = fieldSpec.getDataType().size();
+      break;
+    default:
+      throw new UnsupportedOperationException(
+          "Schema evolution not supported for data type:" + fieldSpec.getDataType());
+    }
+    boolean createDictionary = true;
+    boolean isSortedColumn = createDictionary;
+    boolean hasNulls = false;
+    int totalNumberOfEntries = 0;
```

Comment:
yes, currently we don't use that for single value fields. Was not sure of the implication if I fix it. Will fix it anyway :)